include/mbedtls/pk.h

```diff
@@ -484,7 +484,7 @@ int mbedtls_pk_can_do_ext(const mbedtls_pk_context *ctx, psa_algorithm_t alg,
                           psa_key_usage_t usage);
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
-#if defined(MBEDTLS_PSA_CRYPTO_CLIENT)
+#if defined(MBEDTLS_PSA_CRYPTO_C)
```

Comment:
Are you still planning to do the improvement on this (i.e. make it available when CRYPTO_CLIENT && !CRYPTO_C) in time for 3.6? Otherwise integration that will want to use this client/service separation will have to patch this separately. As those will be both LTS releases, it will be a patched integration that will have to survive for several months at least.

Reply:
I'll try in a follow-up, yes. Perhaps after other work from Valerio that affects the existing code that needs to change.

Reply:
@mpg  @valeriosetti Do we have an issue generically about making `pk.c` work with `MBEDTLS_PSA_CRYPTO_CLIENT && !…_C`? There's https://github.com/Mbed-TLS/mbedtls/issues/8710 about one specific function, but what about the rest?

Reply:
As far as I remember no, there is no other issue for `CRYPTO_CLIENT && !CRYPTO_C` in the near future (but I'm not aware of medium/long term plans, of course). I worked on #8607 recently, but that's the only one I know about.

Reply:
I wasn't aware this was a goal for 3.6, so I have not been creating issues to track this so far.

I think if we set a low bar as in #8710 (basically that things should build with `CRYPTO_CLIENT && !CRYPTO_C`, with all the required functions present, but tests still depending on `CRYPTO_C`) that sounds probably doable.